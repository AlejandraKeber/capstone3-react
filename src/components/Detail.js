import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import GoBack from './GoBack';
import { fetchDetail } from '../redux/DetailSlice';
import '../styles/Detail.css';

const Detail = () => {
  const { id } = useParams();
  const dispatch = useDispatch();

  const { detail, loading, error } = useSelector((state) => state.detail);

  useEffect(() => {
    dispatch(fetchDetail(id));
  }, [dispatch, id]);

  return (
    <div>
      {loading && <h1 className="fallback">Loading...</h1>}
      {error && <h1 className="fallback">{error}</h1>}
      {detail && (
        <div>
          <GoBack />
          <div>
            <div className="detail-container" key={detail.id}>
              <div className="detail-title">
                <img className="detail-img" src={detail.icon} alt={detail.id} />
                <h2>{detail.name}</h2>
              </div>
              <ul className="detail-info">
                <li className="detail-list">
                  Name:
                  {' '}
                  {detail.name}
                </li>
                <li className="detail-list">
                  Symbol:
                  {' '}
                  {detail.symbol}
                </li>
                <li className="detail-list">
                  Rank:
                  {' '}
                  {detail.rank}
                </li>
                <li className="detail-list">
                  Price:
                  {' '}
                  {detail.price}
                  {' '}
                  $
                </li>
                <li className="detail-list">
                  Price in BTC:
                  {' '}
                  {detail.priceBtc}
                </li>
                <li className="detail-list">
                  Available Supply:
                  {' '}
                  {detail.availableSupply}
                </li>
                <li className="detail-list">
                  Total Supply:
                  {' '}
                  {detail.totalSupply}
                </li>
                <li className="detail-list">
                  Price Change in hour:
                  {' '}
                  {detail.priceChange1h}
                </li>
                <li className="detail-list">
                  Price Change in day:
                  {' '}
                  {detail.priceChange1d}
                </li>
                <li className="detail-list">
                  Price Change in week:
                  {' '}
                  {detail.priceChange1w}
                </li>
              </ul>
            </div>
          </div>
        </div>
      )}
    </div>
  );
};
export default Detail;
