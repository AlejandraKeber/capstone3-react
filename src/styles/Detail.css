.detail-container {
  padding: 4rem;
}

.detail-title {
  text-align: center;
  padding: 2rem;
}

li {
  list-style: none;
  font-size: larger;
  padding: 1rem;
}

.detail-list:nth-child(odd) {
  background-color: #334e8e;
}
