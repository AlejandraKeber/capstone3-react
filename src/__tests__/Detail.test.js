import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../redux/configureStore';
import Detail from '../components/Detail';

describe('Detail component', () => {
  it('Rendered Detail component', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Detail />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});