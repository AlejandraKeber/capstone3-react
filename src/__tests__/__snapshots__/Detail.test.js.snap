// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component Rendered Detail component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="fallback"
        >
          Loading...
        </h1>
        <div>
          <div>
            <button
              type="button"
            >
              <h3>
                Go Back
              </h3>
            </button>
          </div>
          <div>
            <div
              class="detail-container"
            >
              <div
                class="detail-title"
              >
                <img />
                <h2 />
              </div>
              <ul
                class="detail-info"
              >
                <li
                  class="detail-list"
                >
                  Name:
                   
                </li>
                <li
                  class="detail-list"
                >
                  Symbol:
                   
                </li>
                <li
                  class="detail-list"
                >
                  Rank:
                   
                </li>
                <li
                  class="detail-list"
                >
                  Price:
                   
                   
                  $
                </li>
                <li
                  class="detail-list"
                >
                  Price in BTC:
                   
                </li>
                <li
                  class="detail-list"
                >
                  Available Supply:
                   
                </li>
                <li
                  class="detail-list"
                >
                  Total Supply:
                   
                </li>
                <li
                  class="detail-list"
                >
                  Price Change in hour:
                   
                </li>
                <li
                  class="detail-list"
                >
                  Price Change in day:
                   
                </li>
                <li
                  class="detail-list"
                >
                  Price Change in week:
                   
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="fallback"
      >
        Loading...
      </h1>
      <div>
        <div>
          <button
            type="button"
          >
            <h3>
              Go Back
            </h3>
          </button>
        </div>
        <div>
          <div
            class="detail-container"
          >
            <div
              class="detail-title"
            >
              <img />
              <h2 />
            </div>
            <ul
              class="detail-info"
            >
              <li
                class="detail-list"
              >
                Name:
                 
              </li>
              <li
                class="detail-list"
              >
                Symbol:
                 
              </li>
              <li
                class="detail-list"
              >
                Rank:
                 
              </li>
              <li
                class="detail-list"
              >
                Price:
                 
                 
                $
              </li>
              <li
                class="detail-list"
              >
                Price in BTC:
                 
              </li>
              <li
                class="detail-list"
              >
                Available Supply:
                 
              </li>
              <li
                class="detail-list"
              >
                Total Supply:
                 
              </li>
              <li
                class="detail-list"
              >
                Price Change in hour:
                 
              </li>
              <li
                class="detail-list"
              >
                Price Change in day:
                 
              </li>
              <li
                class="detail-list"
              >
                Price Change in week:
                 
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
